docs/core-concepts/route-params.md

```diff
@@ -0,0 +1,156 @@
+# Route Params
+
+When defining your route `path`, you can use a colon `:` to denote dynamic params.
+
+```ts
+import { 
+  Routes,
+} from 'kitbag/router'
+
+const routes = [
+  {
+    name: 'users',
+    path: '/users/:id',
+    component: ...
+  }
+] as const satisfies Routes
+```
+
+This means the route will expect 1+ extra string characters (anything but a forward slash `/`) in order to be considered a match. The value of these extra characters is captured and exposed in the `route.params`
+
+```ts
+const route = useRoute()
+
+route.params.id
+```
+
+When defined this way, these params are reactive strings. If you update a param, the url will be updated accordingly.
+
+## Param Types
+
+With the `path` function, kitbag/router supports parsing params to types other than `string`.
+
+```ts
+import { 
+  Routes,
+  path, // [!code ++]
+} from 'kitbag/router'
+
+const routes = [
+  {
+    name: 'users',
+    path: '/users/:id', // [!code --]
+    path: path('/users/:id', { id: Number }), // [!code ++]
+    component: ...
+  }
+] as const satisfies Routes
+```
+
+This will automatically parse the param from `string` in the URL to `number` in `route.params`. If the value cannot be parsed, the route will not be considered a match.
+
+kitbag/router ships with support for `String` (default), `Boolean`, `Number`, and `RegExp`.
+
+### RegExp Params
+
+Using native RegExp is a powerful way of controlling route matching.
+
+```ts
+const routes = [
+  {
+    name: 'users',
+    path: path('/users/:id', { id: /^A[0-9]$/  }), // [!code focus]
+    component: ...
+  }
+] as const satisfies Routes
+```
+
+### Custom Param
+
+You're not limited to the param types that ship with kitbag/router, use `ParamGetter<T>` or `ParamGetSet<T>` to parse params to whatever type you need.
```

Comment:
I think after the intro we should just refer to this as "router" rather than "kitbag/router". But I don't feel strongly. So keep it if you like it.

Reply:
I had a similar thought but decided to keep kitbag there because vue-router always refers to their router as "Vue Router", tanstack always refers to their roouter as "Tanstack Router", just seems like the norm. 

I would be open to using "Kitbag Router" instead of "kitbag/router" if you like that more?

Reply:
IMO that's more readable 👍